### PR TITLE
Fix bugs & add logs 

### PR DESCRIPTION
**Pull request by szabado**
Add a bunch of logs and patch some bugs in the mysql persister that were exposed by those logs

### CODE REVIEW

File: config/mysqlpersister/mysqlpersister.go [hunk 1]
```diff
@@ -54,21 +58,30 @@ func New(c Connector, pollingInterval time.Duration) (*MysqlPersister, error) {
...
 	if _, err := mp.pullConfigs(); err != nil {
 		return nil, err
 	}
 
+	mp.m.RLock()
+	v := mp.latestVersion
+	mp.m.RUnlock()
+	logging.Printf("Pulling configs from MySQL: OK; Latest Version: %v", v)
+
+	mp.notifyWatcher()
```
Comment (szabado) [inline config/mysqlpersister/mysqlpersister.go]: Bug fix

File: config/mysqlpersister/mysqlpersister.go [hunk 2]
```diff
@@ -23,7 +24,7 @@ type MysqlPersister struct {
 	db            *sql.DB
 	m             *sync.RWMutex
 
-	watcher         chan struct{}
+	notifier        *internal.Notifier
```
Comment (szabado) [inline config/mysqlpersister/mysqlpersister.go]: Bug fix

File: config/internal/notifier.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package config
+package internal
```
Comment (szabado) [inline config/internal/notifier.go]: Moved this here to reduce exposed API and break a dependency cycle

File: config/mysqlpersister/mysqlpersister.go [hunk 4]
```diff
@@ -164,18 +185,20 @@ func (mp *MysqlPersister) PersistAndNotify(_ string, c *qsc.ServiceConfig) error
 
 	_, err = mp.db.Exec(q, args...)
 	if err != nil {
-		mysqlErr, ok := err.(*mysql.MySQLError)
-		if ok && mysqlErr.Number == 1062 {
+		if mysqlErr, ok := err.(*mysql.MySQLError); ok && mysqlErr.Number == 1062 {
```
Comment (spenceral) [inline config/mysqlpersister/mysqlpersister.go]: nit: I think it would be neat to make `1062` a const with a descriptive name.

File: server.go [hunk 5]
```diff
@@ -62,16 +62,27 @@ func (s *server) Start() (bool, error) {
 		}
 	}, bufSize)
 
+	logging.Printf("Creating bucket container")
```
Comment (spenceral) [inline server.go]: Do we need this logging (as in the added logging generally, not this line in particular) long term?

Comment (szabado) [reply server.go]: I think it's not a bad thing to have more logging, but some of these bits are a little excessive.

Maybe we can put the excessively verbose log lines at a trace level when we add leveled logging?

File: config/mysqlpersister/mysqlpersister.go (hunk 4 above)
Comment (szabado) [reply config/mysqlpersister/mysqlpersister.go]: Good idea. Did it

**Pull request merged**